ion/event.go

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ion
+
+import (
+	"math/big"
+	"time"
+)
+
+type Event interface {
+	// Returns the depth into the Ion value that this Reader has traversed. At top level the depth is 0, and
+	// it increases by one on each call to StepIn().
+	Depth() int
+
+	// Returns the symbol table that is applicable to the current value.
+	SymbolTable() SymbolTable
+
+	// Returns the type of the current value, or null if there is no current value.
+	Type() Type
+
+	// Return the annotations of the current value as an array of strings.
+	TypeAnnotations() []string
+
+	// Returns the current value's annotations as symbol tokens (text + ID).
+	TypeAnnotationSymbols() []SymbolToken
+
+	// Determines if the current value contains such annotation.
+	HasAnnotation(annotation string) bool
+
+	// Return the field name of the current value, or null if there is no current value.
+	FieldName() string
+
```

Comment:
Given that the API here provides operations to test first if something is there (or the same type) I would expect the "getter" methods to always return with errors. I am not sure if this is the reccommended Go pattern though ...